lib/model.js

```diff
@@ -146,6 +146,21 @@ Model.prototype.$isMongooseModelPrototype = true;
 
 Model.prototype.db;
 
+/**
+ * @api public
+ */
+
+Model.useConnection = function useConnection(connection) {
+  if (!connection) {
+    throw new Error('Please provide a connection.');
+  }
+  if (this.db) {
+    delete this.db.models[this.modelName];
+  }
+  this.db = connection;
+  connection.models[this.modelName] = this;
```

Comment:
We'll need to test this a bit more thoroughly, it looks like we also need to modify the model's collection as well